### PR TITLE
[Ability] Remove Poison Puppeteer's species lock

### PR DESCRIPTION
**Pull request by innerthunder**
## What are the changes?

This removes the condition that makes Poison Puppeteer exclusive to Pecharunt, as well as removing the reference to Pecharunt in the ability's English description.

## Why am I doing these changes?

Poison Puppeteer is currently exclusive to Pecharunt. This causes problems in this game for the following reasons:

- When another Pokemon fuses with Pecharunt, Poison Puppeteer is disabled.
- From a balance perspective, Poison Puppeteer cannot be used as a passive ability for other Pokemon even when it would be a major benefit to them.

Considering Poison Puppeteer doesn't introduce any mechanics that require its source to be Pecharunt, it doesn't make sense to keep it exclusive to the species.

## What did change?

- `data/ability`: Poison Puppeteer's `ConfusionOnStatusEffectAbAttr` is no longer conditional to the source's species.
- `locale/en/ability`: Poison Puppeteer's description no longer references Pecharunt, using "this Pokémon" in its place.

### Screenshots/Videos

Test Video:
- Bulbasaur was given Poison Puppeteer via `src/overrides.ts`.
- The video starts by showing Poison Puppeteer's new English description in Bulbasaur's summary.
- Bulbasaur uses Toxic to poison the opposing Caterpie. With Poison Puppeteer activating, this also confuses Caterpie.

https://github.com/pagefaultgames/pokerogue/assets/168692175/a4b86616-41f1-410c-8b68-fb288388fb0e

## How to test the changes?

See the description for the test video above.

## Checklist
- [x] There is no overlap with another PR?
- [x] The PR is self-contained and cannot be split into smaller PRs?
- [x] Have I provided a clear explanation of the changes?
- [x] Have I tested the changes (manually)?
    - [x] Are all unit tests still passing? (`npm run test`)
- [x] Are the changes visual?
  - [x] Have I provided screenshots/videos of the changes?

### CODE REVIEW

**Comment by CodeTappert** [comment]
``German: 

from
```
description: "Wenn Infamomo ein Ziel mit einer Attacke vergiftet, so wird dieses auch verwirrt.",
```
To
```
description: "Wenn der Anwender ein Ziel mit einer Attacke vergiftet, so wird dieses auch verwirrt.",
```

**Comment by innerthunder** [comment]
Updated French locale:

from

```
description: "Lorsque Pêchaminus empoisonne un Pokémon grâce à l’une de ses capacités, ce dernier devient également confus.",
```

to

```
description: "Lorsque ce Pokémon empoisonne un Pokémon grâce à l’une de ses capacités, ce dernier devient également confus.",
```

**Comment by InnocentGameDev** [comment]
Spanish locale:

from

    description: "Los rivales que Pecharunt envenene con sus movimientos también sufrirán confusión."

to

    description: "Los rivales que este Pokémon envenene con sus movimientos también sufrirán confusión."

**Review by OrangeRed — APPROVED** [review]
Logically everything looks good. The rest is whether the move should be balanced this way.

**Review by Tempo-anon — DISMISSED** [review]
Code looks good to me. Will just leave it up to the lead devs for how we want to deal with species locking abilities in the future

**Comment by innerthunder** [comment]
Just rebased this to fix merge conflicts.

For what it's worth, I had a conversation with @Madmadness65 about this a while ago:

image

**Comment by OrangeRed** [comment]
@innerthunder can you add localization for the other language or at minimum at the key to other locales.

File: src/locales/pt_BR/ability.ts [hunk 1]
```diff
@@ -1235,6 +1235,6 @@ export const ability: AbilityTranslationEntries = {
   },
   poisonPuppeteer: {
     name: "Poison Puppeteer",
-    description: "Pokémon envenenados pelos movimentos de Pecharunt também ficarão confusos.",
+    description: "Pokémon envenenados pelos movimentos de Pecharunt também ficarão confusos.", // TODO: Remove reference to Pecharunt
```
Comment (josericardo-fo) [inline src/locales/pt_BR/ability.ts:1238]:
```suggestion
    description: "Pokémon envenenados pelos movimentos desse Pokémon também ficarão confusos.",
```

File: src/locales/zh_CN/ability.ts [hunk 2]
```diff
@@ -1239,6 +1239,6 @@ export const ability: AbilityTranslationEntries = {
   },
   poisonPuppeteer: {
     name: "毒傀儡",
-    description: "因桃歹郎的招式而陷入中毒状态的\n对手同时也会陷入混乱状态。",
+    description: "因桃歹郎的招式而陷入中毒状态的\n对手同时也会陷入混乱状态。", // TODO: Remove reference to Pecharunt
```
Comment (sonnyding1) [inline src/locales/zh_CN/ability.ts:1242]:
```suggestion
    description: "因为此宝可梦的招式而陷入中毒状\n态的对手同时也会陷入混乱状态。",
```

File: src/locales/zh_TW/ability.ts [hunk 3]
```diff
@@ -1141,6 +1141,6 @@ export const ability: AbilityTranslationEntries = {
   poisonPuppeteer: {
     name: "毒傀儡",
     description:
-            "因為桃歹郎的招式而陷入中\n毒狀態的對手同時也會陷入\n混亂狀態。",
+            "因為桃歹郎的招式而陷入中\n毒狀態的對手同時也會陷入\n混亂狀態。", // TODO: Remove reference to Pecharunt
```
Comment (sonnyding1) [inline src/locales/zh_TW/ability.ts:1144]:
```suggestion
            "因為此寶可夢的招式而陷入中毒狀\n態的對手同時也會陷入混亂狀態。",
```

**Review by sonnyding1 — COMMENTED** [review]
Add zh-cn and zh-tw translations

File: src/locales/ko/ability.ts [hunk 4]
```diff
@@ -1247,6 +1247,6 @@ export const ability: AbilityTranslationEntries = {
   },
   poisonPuppeteer: {
     name: "독조종",
-    description: "복숭악동의 기술에 의해 독 상태가 된 상대는 혼란 상태도 되어 버린다."
+    description: "복숭악동의 기술에 의해 독 상태가 된 상대는 혼란 상태도 되어 버린다." // TODO: Remove reference to Pecharunt
```
Comment (returntoice) [inline src/locales/ko/ability.ts:1250]:
```suggestion
    description: "이 포켓몬의 기술에 의해 독 상태가 된 상대는 혼란 상태도 되어 버린다."
```

**Review by returntoice — CHANGES_REQUESTED** [review]

File: src/locales/it/ability.ts [hunk 5]
```diff
@@ -1239,6 +1239,6 @@ export const ability: AbilityTranslationEntries = {
   },
   poisonPuppeteer: {
     name: "	Malia Tossica",
-    description: "I Pokémon avvelenati dalle mosse di Pecharunt entreranno anche in stato di confusione.",
+    description: "I Pokémon avvelenati dalle mosse di Pecharunt entreranno anche in stato di confusione.", // TODO: Remove reference to Pecharunt
```
Comment (EnochG1) [inline src/locales/it/ability.ts:1242]:
```suggestion
    description: "I Pokémon avvelenati dalle mosse di questo Pokémon entreranno anche in stato di confusione.",
```

**Review by EnochG1 — COMMENTED** [review]
@NicusPulcis requested to modify Italian via Discord.

**Comment by innerthunder** [comment]
All translations should be done now, thanks everyone!

**Review by returntoice — APPROVED** [review]

**Review by EnochG1 — APPROVED** [review]

**Comment by Tempo-anon** [comment]
Closing this because of merge conflicts and we still never reached an actual consensus for what to do about species-locked abilities. Such an edge case though and I don't believe the passive team has assigned poison puppeteer anywhere

**Pull request closed without merging**